sdk-core/src/main/java/care/data4life/sdk/BaseClient.kt

```diff
@@ -16,54 +16,66 @@
 package care.data4life.sdk
 
 import care.data4life.sdk.auth.AuthClient
+import care.data4life.sdk.auth.AuthContract
 import care.data4life.sdk.auth.UserService
 import care.data4life.sdk.call.CallHandler
 import care.data4life.sdk.data.DataRecordClient
 import care.data4life.sdk.fhir.Fhir4RecordClient
 import care.data4life.sdk.log.Log
 import care.data4life.sdk.log.Logger
 
-abstract class BaseClient(
```

Comment:
Out out pure curiosity: what was the reasoning behind choosing a private-public constructor pair, rather than sticking with the existing constructor and adding `val userID = userService.userID.blockingGet()` inside the class (or in an `init` block) to initialize the property on construction?

Reply:
Monday Patch - you are totally right for pointing on that, thx.